src/controller/expenses.controller.js

```diff
@@ -0,0 +1,107 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  if (!userId && !categories && (!from || !to)) {
+    res.send(expensesService.getAllExpenses());
+  }
+
+  const normalizedCategories = expensesService.normalizeCategory(categories);
+
+  let filteredExpenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(userId);
+  }
+
+  if (normalizedCategories) {
+    filteredExpenses =
+      expensesService.filterExpensesByCategory(normalizedCategories);
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(fromDate, toDate);
+  }
```

Comment:
There is a performance issue here. You are fetching all expenses from your service and then filtering them in the controller. It would be more efficient to perform the filtering in the service itself, as it would reduce the amount of data that needs to be transferred and processed. Consider passing the filter parameters to the service and performing the filtering there.